File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        nb_neg = 0
        for n in nums:
            if not(n):
                return 0
            if n < 0:
                if nb_neg:
                    nb_neg = 0
                else:
                    nb_neg = 1
        if nb_neg:
            return -1
        return 1

    """
    def arraySign(self, nums: List[int]) -> int:
        res = 1
        for n in nums:
            if not(n):
                return 0
            if n < 0:
                res *= -1
        return res
    """

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.rs
impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut nb_neg: i32 = 0;
        for n in &nums {
            if (*n == 0) {
                return 0;
            }
            if (*n < 0) {
                if (nb_neg == 1) {
                    nb_neg = 0;
                }
                else {
                    nb_neg = 1;
                }
            }
        }
        if (nb_neg == 1) {
            return -1;
        }
        return 1;
    }
}
